lib/wasix/src/syscalls/wasix/context_create.rs

```diff
@@ -0,0 +1,96 @@
+use crate::{WasiEnv, WasiError};
+use tracing::instrument;
+use wasmer::{
+    AsStoreRef, Function, FunctionEnvMut, MemorySize, RuntimeError, StoreMut, TypedFunction, Value,
+    WasmPtr,
+};
+use wasmer_wasix_types::wasi::Errno;
+
+/// Return the function corresponding to the given entrypoint index if it exists and has the signature `() -> ()`
+pub fn lookup_typechecked_entrypoint(
+    data: &WasiEnv,
+    mut store: &mut StoreMut<'_>,
+    entrypoint_id: u32,
+) -> Result<TypedFunction<(), ()>, Errno> {
+    let entrypoint = match data
+        .inner()
+        .indirect_function_table_lookup(&mut store, entrypoint_id)
+    {
+        Ok(func) => func,
+        Err(e) => {
+            tracing::trace!(
+                "Failed to lookup entrypoint function {}: {:?}",
+                entrypoint_id,
+                e
+            );
+            return Err(Errno::Inval);
+        }
+    };
+
+    let entrypoint_type = entrypoint.ty(&store);
+    let Ok(entrypoint) = entrypoint.typed::<(), ()>(&store) else {
```

Comment:
Why `() -> ()`? Don't we pass give the new context its ID or something?

Reply:
No, because we need the new context ID immediately to return it from the function. As this API only provides a minimal interface for context switching, the guest code can pass that ID to the new context in some other way if it wants to.